### ISSUE TITLE
Hangs and crashes with 1.4.2 - WAS: Regression between 1.4.1 and >= 1.4.2 in `k8s.yaml.ConfigGroup`

### ISSUE THREAD

**Issue by ohardy** [pulumi/pulumi-kubernetes#963]
### Problem description

In 1.4.1 we create a k8s.yaml.ConfigGroup with 10 yaml files and everything was ok. When I migrate to 1.4.2 or to last version 1.4.5, `pulumi up` just freeze while generating the preview state. If I comment ConfigGroup creation, everything works again and If I go back to version 1.4.1 everything works again too.

### Affected product version(s)

>= 1.4.2

### Reproducing the issue

The problem appear with a simple yaml from cert-manager for exemple: https://github.com/jetstack/cert-manager/releases/download/v0.12.0/cert-manager.yaml and this code:
```
const configGroup = new k8s.yaml.ConfigGroup(
    'kubernetes-ressources',
    {
      files: [
        '../manifests/cert-manager.yaml',
      ],
    },
    { provider: clusterProvider }
  );
```

Any idea ?

**Comment by lukehoban** [comment]
The hang - especially after moving to use invokes in the releases in question  - sounds like it could be related to NodeJS hangs with deasync. But the new codepaths here use `{async: true}` which should avoid that. 

Cc @lblackstone @CyrusNajmabadi

**Comment by lblackstone** [comment]
I'm observing the same behavior with this manifest, and am wondering if it's related to the file size. This YAML file is ~6.4k LOC, so it's possible the "hang" is actually a performance issue with `invoke`.

**Comment by lblackstone** [comment]
I left it running awhile longer this time, and the preview eventually crashed with this error:
```
  pulumi:pulumi:Stack (pulumi-k8s-test-pulumi-k8s-test-dev):
    FATAL ERROR: invalid array length Allocation failed - JavaScript heap out of memory
     1: 0x10007f231 node::Abort() [/usr/local/bin/node]
     2: 0x10007f3b5 node::OnFatalError(char const*, char const*) [/usr/local/bin/node]
     3: 0x100176887 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/usr/local/bin/node]
     4: 0x100176823 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/usr/local/bin/node]
     5: 0x1002fa9d5 v8::internal::Heap::FatalProcessOutOfMemory(char const*) [/usr/local/bin/node]
     6: 0x1002d8b25 v8::internal::Factory::CopyWeakArrayListAndGrow(v8::internal::Handle<v8::internal::WeakArrayList>, int, v8::internal::AllocationType) [/usr/local/bin/node]
     7: 0x1004f3f81 v8::internal::PrototypeUsers::Add(v8::internal::Isolate*, v8::internal::Handle<v8::internal::WeakArrayList>, v8::internal::Handle<v8::internal::Map>, int*) [/usr/local/bin/node]
     8: 0x1004ac037 v8::internal::JSObject::LazyRegisterPrototypeUser(v8::internal::Handle<v8::internal::Map>, v8::internal::Isolate*) [/usr/local/bin/node]
     9: 0x1004d2588 v8::internal::Map::GetOrCreatePrototypeChainValidityCell(v8::internal::Handle<v8::internal::Map>, v8::internal::Isolate*) [/usr/local/bin/node]
    10: 0x1003846de v8::internal::LoadHandler::LoadFromPrototype(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Map>, v8::internal::Handle<v8::internal::JSReceiver>, v8::internal::Handle<v8::internal::Smi>, v8::internal::MaybeObjectHandle, v8::internal::MaybeObjectHandle) [/usr/local/bin/node]
    11: 0x100389bbf v8::internal::LoadIC::ComputeHandler(v8::internal::LookupIterator*) [/usr/local/bin/node]
    12: 0x1003889ac v8::internal::LoadIC::UpdateCaches(v8::internal::LookupIterator*) [/usr/local/bin/node]
    13: 0x100388586 v8::internal::LoadIC::Load(v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Name>) [/usr/local/bin/node]
    14: 0x10038f896 v8::internal::Runtime_LoadIC_Miss(int, unsigned long*, v8::internal::Isolate*) [/usr/local/bin/node]
    15: 0x1009311f9 Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_NoBuiltinExit [/usr/local/bin/node]

    <--- Last few GCs --->
    [18019:0x102892000]  1113821 ms: Scavenge 2171.2 (2193.6) -> 2159.2 (2195.1) MB, 6.2 / 0.0 ms  (average mu = 0.989, current mu = 0.989) allocation failure
    [18019:0x102892000]  1113848 ms: Scavenge 2173.3 (2195.6) -> 2161.9 (2196.8) MB, 6.3 / 0.0 ms  (average mu = 0.989, current mu = 0.989) allocation failure
    [18019:0x102892000]  1113878 ms: Scavenge 2175.3 (2197.3) -> 2163.0 (2199.3) MB, 6.4 / 0.0 ms  (average mu = 0.989, current mu = 0.989) allocation failure
    <--- JS stacktrace --->
    ==== JS stack trace =========================================
        0: ExitFrame [pc: 0x1009311f9]
    Security context: 0x19ae61d408a1 <JSObject>
        1: /* anonymous */(aka /* anonymous */) [0x19aec3493e31] [/Users/levi/workspace/pulumi-k8s-test/node_modules/@pulumi/kubernetes/node_modules/@pulumi/pulumi/output.js:~17] [pc=0x2dae01e98193](this=0x19ae205404a9 <undefined>,0x19aec3493ee9 <JSFunction (sfi = 0x19aeafe0c271)>,0x19aec3493f21 <JSFunction (sfi = 0x19aeafe0c2a9)>)
        2: new Promise(aka Promise) [...

    error: an unhandled error occurred: Program exited with non-zero exit code: -1
```

**Comment by arnediekmann** [comment]
As a side note: we had the same problem but could fix it at first by downgrading `@pulumi/kubernetes` to `1.4.1`. However, after upgrading `@pulumi/pulumi` to `1.9.1` it started hanging again (only when trying to deploy the cert-manager as described in #964) - Downgrading that package to `1.9.0` resolved the issue again.

**Comment by julienvincent** [comment]
Just to add another datapoint - I am running into this as well while trying to apply a `Velero` stack's CRD's (~1.5k loc).

Reverting to kubernetes `1.4.1` and pulumi `1.9.0` resolved it for me as well

**Comment by lblackstone** [comment]
We will be cutting a release with the fix soon, but it's merged to master, and you can try it out in the meantime with `pulumi-kubernetes: dev` in `package.json`

**Comment by jpmckearin** [comment]
@lblackstone was this released with v1.5.0?

**Comment by lblackstone** [comment]
This was released in `v1.5.0`
